cl/search/tests/test_pacer_bulk_fetch.py

```diff
@@ -0,0 +1,640 @@
+        # Court 0: Many large docs
+        for i in range(300):
+            self.docs.append(
+                RECAPDocumentFactory(
+                    docket_entry=self.docket_entries[0][i],
+                    pacer_doc_id=f"0_{i}",
+                    is_available=False,
+                    page_count=1000 + i,
+                )
+            )
+
+        # Court 1: Mix of pages
+        page_counts = [5, 10, 50, 100, 500, 1000, 2000, 5000]
+        for i, pages in enumerate(page_counts):
+            self.docs.append(
+                RECAPDocumentFactory(
+                    docket_entry=self.docket_entries[1][i],
+                    pacer_doc_id=f"1_{i}",
+                    is_available=False,
+                    page_count=pages,
+                )
+            )
+
+        # Court 2: Only small docs (1-10 pages)
+        for i in range(20):
+            self.docs.append(
+                RECAPDocumentFactory(
+                    docket_entry=self.docket_entries[2][i],
+                    pacer_doc_id=f"2_{i}",
+                    is_available=False,
+                    page_count=i + 1,
+                )
+            )
+
+        # Court 3: Only medium docs (100-500 pages)
+        for i in range(15):
+            self.docs.append(
+                RECAPDocumentFactory(
+                    docket_entry=self.docket_entries[3][i],
+                    pacer_doc_id=f"3_{i}",
+                    is_available=False,
+                    page_count=100 + (i * 25),
+                )
+            )
+
+        # Court 4: No docs matching typical filters (all 11-49 pages)
+        for i in range(10):
+            self.docs.append(
+                RECAPDocumentFactory(
+                    docket_entry=self.docket_entries[4][i],
+                    pacer_doc_id=f"4_{i}",
+                    is_available=False,
+                    page_count=11 + (i * 3),
+                )
+            )
+
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.should_skip",
+        return_value=False,
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.docs_to_process_cache_key",
+        return_value="pacer_bulk_fetch.test_page_count_filtering.docs_to_process",
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.Command.timed_out_docs_cache_key",
+        return_value="pacer_bulk_fetch.test_page_count_filtering.timed_out_docs",
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.append_value_in_cache",
+        wraps=append_value_in_cache,
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.CeleryThrottle.maybe_wait"
+    )
+    @patch("cl.search.management.commands.pacer_bulk_fetch.time.sleep")
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.get_or_cache_pacer_cookies"
+    )
+    @patch(
+        "cl.search.management.commands.pacer_bulk_fetch.fetch_pacer_doc_by_rd.si"
+    )
```

Comment:
I see that many of these mocks are shared across different tests in the class. You can move these common mocks (as long as they don’t require a specific `return_value` or method unique to a test) to the class level. This would help reduce duplicated code across test methods.